server/src/main/java/org/opensearch/cluster/routing/IndexShardRoutingTable.java

```diff
@@ -949,20 +943,55 @@ public int hashCode() {
         }
     }
 
+    @PublicApi(since = "2.14.0")
+    public static class WeightedShardRoutings {
```

Comment:
nit: We can drop the public access modifier?

Reply:
It needs to be public to be accessed in the tests.